sound/soc/intel/boards/sof_rt5682.c

```diff
@@ -683,13 +691,26 @@ static int sof_rt5682_remove(struct platform_device *pdev)
 	return 0;
 }
 
+static const struct platform_device_id board_ids[] = {
+	{
+		.name = "tgl_max98357a_rt5682",
+		.driver_data = (kernel_ulong_t)(SOF_RT5682_MCLK_EN |
+					SOF_RT5682_SSP_CODEC(0) |
+					SOF_SPEAKER_AMP_PRESENT |
+					SOF_RT5682_SSP_AMP(1) |
+					SOF_RT5682_NUM_HDMI_DAI_FOUR),
+	},
+	{ }
```

Comment:
Is the other platform use rt5682 need to fill this table?

From CI test result, BYT RT5682 machine driver load is blocked.
https://sof-ci.01.org/linuxpr/PR1505/build2489/devicetest/BYT_MB_RT5682/verify-pcm-list.sh/dmesg.txt
```
[   18.342601 <    0.000357>] sof-audio-acpi 80860F28:01: ipc tx succeeded: 0x90030000: GLB_TRACE_MSG
[   18.342745 <    0.000144>] sof-audio-acpi 80860F28:01: created machine sof_rt5682
[   18.843475 <    0.500730>] sof-audio-acpi 80860F28:01: ipc rx: 0x90020000: GLB_TRACE_MSG
[   18.843487 <    0.000012>] sof-audio-acpi 80860F28:01: ipc rx done: 0x90020000: GLB_TRACE_MSG
[   20.769785 <    1.926298>] random: crng init done
[   20.769790 <    0.000005>] random: 7 urandom warning(s) missed due to ratelimiting
[   26.222744 <    5.452954>] systemd-journald[230]: File /var/log/journal/e396fab747714ec582f3775562507c6e/user-1000.journal corrupted or uncleanly shut down, renaming and replacing.
[   27.170120 <    0.947376>] igb 0000:03:00.0 enp3s0: igb: enp3s0 NIC Link is Up 100 Mbps Full Duplex, Flow Control: RX
[   27.170356 <    0.000236>] IPv6: ADDRCONF(NETDEV_CHANGE): enp3s0: link becomes ready
[   39.407405 <   12.237049>] rfkill: input handler disabled
[  115.215351 <   75.807946>] sof-audio-acpi 80860F28:01: Copying cached debugfs data
```

Reply:
>Is the other platform use rt5682 need to fill this table?
No. Not needed.
For Byt, sof_rt5682_quirk is getting assigned here in the same sof_audio_probe():

if (soc_intel_is_byt() || soc_intel_is_cht()) {
....
		/* default quirk for legacy cpu */
		sof_rt5682_quirk = SOF_RT5682_MCLK_EN |
						SOF_RT5682_MCLK_BYTCHT_EN |
						SOF_RT5682_SSP_CODEC(2);
Thinking what could be causing issue...

Reply:
@sathya-nujella from the logs, it doesnt look like the machine driver is even getting probed. 
@fredoh9 what does this error mean? 
"2019-11-18 01:40:46 UTC [REM_WARNING]: The /lib/firmware/intel/sof-tplg/sof-byt-rt5682.tplg doesn't match the loaded topology!"

Reply:
@sathya-nujella I have a feeling we need a entry for sof-rt5682 in the board_ids table like this:
{ .name = "sof_rt5682" },

Reply:
@ranj063 this error is printed when input tplg file in test framework doesn't reflect currently loaded tplg. sof-byt-rt5682.tplg is right one?

Reply:
> @sathya-nujella I have a feeling we need an entry for sof-rt5682 in the board_ids table like this:
> { .name = "sof_rt5682" },

Yes, this may be the cause.. in that case, it should affect CML Hatch too. Let me cross check on it as I don't have BYT. Thanks Ranjani for the tips!

Reply:
I am able to reproduce issue in Hatch similar to the one failed in BYT. If platform driver structure includes id_table, then it will go through all the entries of id_table for matching name. I missed adding original "sof_rt5682". So, it did not work. Yes, as Ranjani mentioned that is the issue. sorry for late in getting this done.

Reply:
The issue in BYT is solved with new patchset. So, resolving conversation.